src/peer_digest.cc

```diff
@@ -344,12 +331,16 @@ peerDigestHandleReply(void *data, StoreIOBuffer receivedData)
     digest_read_state_t prevstate;
     int newsize;
 
+    if (!fetch->pd.valid() || !fetch->pd->peer.valid()) {
```

Comment:
If pd is valid (i.e. the first condition is false), then pd->peer must be valid as well (i.e. the second condition must be false as well) because CachePeer destruction invalidates the digest object as well), right?

```suggestion
    if (!fetch->pd.valid()) {
```